### PR TITLE
feature: enabling freethreaded python support for pypi/conda

### PR DESCRIPTION
**Pull request by saikonen**
UX and possible corner cases still to be covered. Current approach is to have a `t` suffix in the python version to denote no GIL

### CODE REVIEW

**Comment by saikonen** [comment]
easy test flow (swap out `@conda` for `@pypi` to test abi tag support):

```python
from metaflow import step, FlowSpec, conda
import sys

class CondaGILFlow(FlowSpec):
    @conda(python="3.13t")
    @step
    def start(self):
        print("Starting 👋")
        print("Is GIL? ", sys._is_gil_enabled())
        self.next(self.end)
    
    @conda(python="3.14t")
    @step
    def end(self):
        print("Is GIL? ", sys._is_gil_enabled())
        print("Done! 🏁")


if __name__ == "__main__":
    CondaGILFlow()
```

File: metaflow/plugins/pypi/pip.py [hunk 1]
```diff
@@ -83,12 +83,13 @@ def solve(self, id_, packages, python, platform):
             )
 
         debug.conda_exec("Solving packages for PyPI environment %s" % id_)
+        freethreaded = python.endswith("t")
```
Comment (savingoyal) [inline metaflow/plugins/pypi/pip.py:86]: does it work with patch versions?
